unblob/extractor.py

```diff
@@ -15,21 +15,13 @@
 logger = get_logger()
 
 
-APPEND_NAME = "_extract"
-
-
-def make_extract_dir(root: Path, path: Path, extract_root: Path) -> Path:
-    """Create extraction dir under root with the name of path."""
-    relative_path = path.relative_to(root)
-    extract_name = relative_path.name + APPEND_NAME
-    extract_dir = extract_root / relative_path.with_name(extract_name)
-    extract_dir.mkdir(parents=True, exist_ok=True)
-    logger.debug("Created extraction dir", path=extract_dir)
-    return extract_dir.expanduser().resolve()
-
-
 def carve_chunk_to_file(carve_path: Path, file: io.BufferedIOBase, chunk: Chunk):
     """Extract valid chunk to a file, which we then pass to another tool to extract it."""
+    assert carve_path.is_absolute()
```

Comment:
I would prefer not to have assert in a production code, that could result in DoS attacks, rather we should log it and handle it

Reply:
Better, test it. We have the code in place so that it cannot be triggered currently.

Reply:
it is mostly for documentation - poor man's annotation for a pre-condition.

I do not expect it to ever fail, but if it does, we better know about it anyway.

Reply:
removed.